site/app/controllers/admin/PlagiarismController.php

```diff
@@ -68,9 +68,35 @@ private function getGradeablesFromPriorTerm() {
     }
 
 
-    private function getRankings($gradeable_id) {
+    private function getOverallRankings($gradeable_id) {
         $course_path = $this->core->getConfig()->getCoursePath();
-        $file_path = $course_path . "/lichen/ranking/" . $gradeable_id . ".txt";
+        $file_path = $course_path . "/lichen/ranking/" . $gradeable_id . "/overall_ranking.txt";
+        if (!file_exists($file_path)) {
+            $this->core->addErrorMessage("Plagiarism Detection job is running for this gradeable.");
+            $this->core->redirect($this->core->buildCourseUrl(['plagiarism']));
```

Comment:
I think you would want to return at the end of this if statement, otherwise the rest of the function will still execute

Reply:
I'll add a return statement just for good measure but `redirect` dies so the code should never pass this point.

Reply:
On second thought, I'm not sure if adding a return statement is good practice here.  Thoughts?

Reply:
Good practice would be to not have this "helper" function mutate the response state, and rather just return some value, and then the redirect (and `addSuccessMessage` below) would happen in the caller site. The caller could then (eventually) return `RedirectResponse` for this failure case.

For a return value, I would probably suggest doing `: ?array` where if `$rankings === null`, then redirect, if `count($rankings) === 0`, show `successMessage`.

Reply:
Wouldn't it defeat the purpose of having this function to move the error checking back to the caller site though?  The reason I made the function in the first place was to make sure that the same errors were treated the same way, with the same error text, which led to the same place in the code.

Reply:
Perhaps, though pretty usually the callers want to do different things on error conditions. For example, here, I would doubt the `ajaxGetSubmissionConcatenated` based on its name and return type (a stringified JSON object) that having a redirect would be an appropriate error condition.